src/api/app/presenters/involved_items_presenter.rb

```diff
@@ -0,0 +1,131 @@
+class InvolvedItemsPresenter
+  MAX_ITEMS_PER_PAGE = 25
+  KLASS_FILTER = %w[involved_projects involved_packages].freeze
+
+  attr_reader :user, :filters
+
+  def initialize(user:, filters:, page:)
+    @user = user
+    @filters = prepare_filters(filters: filters)
+    @page = page
+  end
+
+  def involved_pkg_and_prj_paginated
+    Kaminari.paginate_array(involved_packages_and_projects).page(@page).per(MAX_ITEMS_PER_PAGE)
+  end
+
+  def involved_packages_and_projects
+    involved_items = []
+    roles = roles_to_filter
+
+    if filter_by_owner?
+      involved_items.concat(involved_items_as_owner)
+    elsif filter_by_role?
+      involved_items.concat(pkg_or_prj_filtered_by_roles(user: @user, klass: Project, roles: roles)) if consider_involved_projects?
+      involved_items.concat(pkg_or_prj_filtered_by_roles(user: @user, klass: Package, roles: roles)) if consider_involved_packages?
+    else
+      involved_items.concat(pkg_or_prj_unfiltered(user: @user, klass: Package)) if consider_involved_packages?
+      involved_items.concat(pkg_or_prj_unfiltered(user: @user, klass: Project)) if consider_involved_projects?
+    end
+
+    involved_items.uniq
+  end
+
+  def owner_root_project_exists?
+    Project.find_by_attribute_type(AttribType.find_by_name!('OBS:OwnerRootProject')).exists?
+  end
+
+  def involved_items_as_owner
+    @involved_items_as_owner ||= []
+
+    return @involved_items_as_owner if @involved_items_as_owner.present?
+
+    @involved_items_as_owner.concat(
+      OwnerSearch::Owned.new.for(@user)
+                        .map { |owner| owned_item(user: owner, search_text: @filters['search_text']) }
+                        .reject(&:nil?)
+    )
+  end
+
+  private
+
+  def prepare_filters(filters:)
+    filters['search_text'] = filters['search_text']&.strip
+    filters
+  end
+
+  def filter_by_role?
+    return true if roles_to_filter.present?
+
```

Comment:
why are you letting so many empty lines between the last expression being evaluated in the methods (aka the return value)?

Reply:
@vpereira that's what Rubocop wanted :) its because of this https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Layout/EmptyLineAfterGuardClause

Reply:
I'm fine with having an empty line or not.